base/sparse/sparsevector.jl

```diff
@@ -688,30 +746,26 @@ function show(io::IOContext, x::AbstractSparseVector)
     nzind = nonzeroinds(x)
     nzval = nonzeros(x)
     xnnz = length(nzind)
-
+    if length(nzind) == 0
+        return show(io, MIME("text/plain"), x)
+    end
     limit::Bool = get(io, :limit, false)
     half_screen_rows = limit ? div(displaysize(io)[1] - 8, 2) : typemax(Int)
     pad = ndigits(n)
-    sep = "\n\t"
     if !haskey(io, :compact)
         io = IOContext(io, :compact => true)
     end
     for k = 1:length(nzind)
         if k < half_screen_rows || k > xnnz - half_screen_rows
             print(io, "  ", '[', rpad(nzind[k], pad), "]  =  ")
             Base.show(io, nzval[k])
```

Comment:
guess this should probably also be
```
            if isassigned(nzval, Int(k))
                 Base.show(io, nzval[k])
             else
                 print(io, Base.undef_ref_str)
            end
```
there just aren't tests of putting a possibly-undef type in as the values of a SparseVector and showing it?